File: backend/app/dto/shared/patient.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field

from app.dto.internal.model_features import ModelFeatures


class GeneVariant(BaseModel):
    model_config = ConfigDict(extra="forbid")

    gene: str
    variant_classification: str


class ClinicalInfo(BaseModel):
    model_config = ConfigDict(extra="forbid")

    age: int | None = None
    pathologic_stage: str | None = None
    gender: str | None = None
    stage: int | None = Field(default=None, exclude=True)


class NormalizedPatientInput(BaseModel):
    model_config = ConfigDict(extra="forbid")

    deidentified_patient_id: str
    gene_variants: list[GeneVariant]
    clinical: ClinicalInfo
    model_features: ModelFeatures | None = Field(default=None, exclude=True)


class PatientReference(BaseModel):
    model_config = ConfigDict(extra="forbid")

    deidentified_patient_id: str

File: frontend/src/features/analysis/model/patient.ts
export type Variant = {
  gene: string;
  variant_classification: string;
};

export type NormalizedClinical = {
  age: number | null;
  pathologic_stage: string | null;
  gender: string | null;
};

export type NormalizedPatientInput = {
  deidentified_patient_id: string;
  gene_variants?: Variant[];
  clinical: NormalizedClinical;
};
